pyccel/ast/class_defs.py

```diff
@@ -0,0 +1,169 @@
+
+StringClass = ClassDef('string',
+        methods=[
+                #capitalize
+                #casefold
+                #center
+                #count
+                #encode
+                #endswith
+                #expandtabs
+                #find
+                #format
+                #format_map
+                #index
+                #isalnum
+                #isalpha
+                #isascii
+                #isdecimal
+                #isdigit
+                #isidentifier
+                #islower
+                #isnumeric
+                #isprintable
+                #isspace
+                #istitle
+                #isupper
+                #join
+                #ljust
+                #lower
+                #lstrip
+                #maketrans
+                #partition
+                #replace
+                #rfind
+                #rindex
+                #rjust
+                #rpartition
+                #rsplit
+                #rstrip
+                #split
+                #splitlines
+                #startswith
+                #strip
+                #swapcase
+                #title
+                #translate
+                #upper
+                #zfill
+                ])
+
+#=======================================================================================
+
+TupleClass = ClassDef('tuple',
+        methods=[
+            #index
+            #count
+            ])
+
+#=======================================================================================
+
+NumpyArrayClass = ClassDef('numpy.ndarray',
+        methods=[
+            FunctionDef('shape',[],[],body=[],
+                decorators={'property':'property', 'numpy_wrapper':Shape}),
+            FunctionDef('sum',[],[],body=[],
+                decorators={'numpy_wrapper':NumpySum}),
+            FunctionDef('min',[],[],body=[],
+                decorators={'numpy_wrapper':NumpyAmin}),
+            FunctionDef('max',[],[],body=[],
+                decorators={'numpy_wrapper':NumpyAmax}),
+            FunctionDef('imag',[],[],body=[],
+                decorators={'property':'property', 'numpy_wrapper':NumpyImag}),
+            FunctionDef('real',[],[],body=[],
+                decorators={'property':'property', 'numpy_wrapper':NumpyReal})])
+
+#=======================================================================================
+
+literal_classes = {
+        NativeBool()    : BooleanClass,
+        NativeInteger() : IntegerClass,
+        NativeReal()    : RealClass,
+        NativeComplex() : ComplexClass,
+        NativeString()  : StringClass
+}
+
+#=======================================================================================
+
+def get_cls_base(dtype, rank):
+    """
+    From the dtype and rank, determine the base class of an object
+    """
+    if rank == 0:
+        return literal_classes[dtype]
+    else:
+        return NumpyArrayClass
```

Comment:
Do you remember if we support Numpy arrays of rank 0?

Reply:
Currently we don't. @rakati was going to change the rank so this was possible (see #836) but this got stuck. If/when it is added this line will need to be changed to `if rank is None`